system/lib/wasmfs/js_api.cpp

```diff
@@ -21,9 +21,11 @@ __wasi_fd_t wasmfs_create_file(char* pathname, mode_t mode, backend_t backend);
 // The buffer will also contain the file length.
 // Caller must free the returned pointer.
 void* _wasmfs_read_file(char* path) {
-  struct stat file;
+  static_assert(sizeof(off64_t) == 8, "File offset type must be 64-bit");
```

Comment:
`off_t` is already 64-bit - could we use that everywhere here? (otherwise this would be the first usage of `off64_t` in wasmfs)

I do think this static assert would be very useful for `off_t`!

Reply:
Ah, great! Commit adc486a1711369d2a14708ee2c1e34452a19dd0b fixes this.

I thought `off_t` was a 32-bit value (as on regular 32-bit systems), because the "old" `FS.read` and `FS.llseek` functions used the JS `Number` type for their `offset` and `position` arguments instead of `BigInt`s, but probably these functions didn't support large files in the first place.

Reply:
Great! lgtm.

Yes, makes sense we had bugs on this in the old code... the type was right and a few code paths were checked, but we never did a comprehensive test on >32 bit file sizes everywhere.